### PR TITLE
fix: fixing mysql error message

### PR DESCRIPTION
**Pull request by AAfghahi**
### SUMMARY
The previous regex for custom database errors for mySQL had an extra period, this fixes that. The same was true for postgres when you are editing an already existing database. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before (postgres):
Screen Shot 2021-04-30 at 3 08 11 PM
After (postgres)
Screen Shot 2021-04-30 at 3 11 55 PM

After
Screen Shot 2021-04-29 at 11 31 15 AM

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/db_engine_specs/mysql.py [hunk 1]
```diff
@@ -42,13 +42,13 @@
 
 # Regular expressions to catch custom errors
 CONNECTION_ACCESS_DENIED_REGEX = re.compile(
-    "Access denied for user '(?P<username>.*?)'@'(?P<hostname>.*?)'. "
+    "Access denied for user '(?P<username>.*?)'@'(?P<hostname>.*?)' "
```
Comment (betodealmeida) [inline superset/db_engine_specs/mysql.py:45]:
```suggestion
    "Access denied for user '(?P<username>.*?)'@'(?P<hostname>.*?)'"
```

**Review by betodealmeida — CHANGES_REQUESTED** [review]

**Comment by betodealmeida** [comment]
@AAfghahi I just realized, can you update the unit tests as well? If though they should pass, it would be nice to have the error message there consistent with what MySQL returns.

**Comment by AAfghahi** [comment]
> @AAfghahi I just realized, can you update the unit tests as well? If though they should pass, it would be nice to have the error message there consistent with what MySQL returns.

Oh yes, good point.

**Comment by AAfghahi** [comment]
@betodealmeida fixed the tests

File: tests/db_engine_specs/mysql_tests.py [hunk 2]
```diff
@@ -196,7 +196,7 @@ def test_extract_errors(self):
             )
         ]
 
-        msg = "mysql: Unknown database 'badDB'."
+        msg = "mysql: Unknown database 'badDB'.ß"
```
Comment (betodealmeida) [inline tests/db_engine_specs/mysql_tests.py:199]:
```suggestion
        msg = "mysql: Unknown database 'badDB'"
```

File: tests/db_engine_specs/mysql_tests.py [hunk 3]
```diff
@@ -135,7 +135,7 @@ def test_extract_errors(self):
             )
         ]
 
-        msg = "mysql: Unknown MySQL server host 'badhostname.com'. "
+        msg = "mysql: Unknown MySQL server host 'badhostname.com' "
```
Comment (betodealmeida) [inline tests/db_engine_specs/mysql_tests.py:138]:
```suggestion
        msg = "mysql: Unknown MySQL server host 'badhostname.com'"
```

File: tests/db_engine_specs/mysql_tests.py [hunk 4]
```diff
@@ -110,7 +110,7 @@ def test_extract_errors(self):
         """
         Test that custom error messages are extracted correctly.
         """
-        msg = "mysql: Access denied for user 'test'@'testuser.com'. "
+        msg = "mysql: Access denied for user 'test'@'testuser.com' "
```
Comment (betodealmeida) [inline tests/db_engine_specs/mysql_tests.py:113]:
```suggestion
        msg = "mysql: Access denied for user 'test'@'testuser.com'"
```

File: superset/db_engine_specs/mysql.py (hunk 1 above)
Comment (betodealmeida) [inline superset/db_engine_specs/mysql.py:45]:
```suggestion
    "Access denied for user '(?P<username>.*?)'@'(?P<hostname>.*?)'"
```

**Review by betodealmeida — CHANGES_REQUESTED** [review]

**Review by betodealmeida — APPROVED** [review]
Awesome!

**Pull request merged**